Makefile

```diff
@@ -138,17 +139,36 @@ properclean: ## Completely clean the repository's build artifacts.
 	rm -rf $(DISTDIR) $(TESTDIR)
 	$(DOCKER) rmi $(IMAGE)
 
-.PHONY: git2go
-git2go: $(VENDORDIR)/github.com/libgit2/git2go/v31/static-build/install/lib/pkgconfig/libgit2.pc
-
-$(VENDORDIR)/github.com/libgit2/git2go/v31/static-build/install/lib/pkgconfig/libgit2.pc: $(VENDORDIR)/github.com/libgit2/git2go/v31/vendor/libgit2
-	$(MAKE) -C $(VENDORDIR)/github.com/libgit2/git2go/v31 install-static
+GIT2GO_VERSION ?= v31
 
-$(VENDORDIR)/github.com/libgit2/git2go/v31/vendor/libgit2: $(VENDORDIR)/github.com/libgit2/git2go
-	$(GIT) -C $(VENDORDIR)/github.com/libgit2/git2go/v31 submodule update --init --recursive
+.PHONY: git2go
+git2go: $(VENDORDIR)/github.com/libgit2/git2go/$(GIT2GO_VERSION)/static-build/install/lib/pkgconfig/libgit2.pc
+	(cd $(VENDORDIR)/github.com/libgit2/git2go/$(GIT2GO_VERSION) && $(GO) install --tags "static" ./...)
+
+$(VENDORDIR)/github.com/libgit2/git2go/$(GIT2GO_VERSION)/static-build/install/lib/pkgconfig/libgit2.pc: $(VENDORDIR)/github.com/libgit2/git2go/$(GIT2GO_VERSION)/vendor/libgit2
+	$(MKDIR) -p $(VENDORDIR)/github.com/libgit2/git2go/$(GIT2GO_VERSION)/static-build/build
+	$(MKDIR) -p $(VENDORDIR)/github.com/libgit2/git2go/$(GIT2GO_VERSION)/static-build/install
+	(cd $(VENDORDIR)/github.com/libgit2/git2go/$(GIT2GO_VERSION)/static-build/build && $(CMAKE) \
+		-DTHREADSAFE=ON \
+		-DBUILD_CLAR=OFF \
+		-DBUILD_SHARED_LIBS=OFF \
+		-DREGEX_BACKEND=builtin \
+		-DUSE_BUNDLED_ZLIB=ON \
+		-DUSE_HTTPS=ON \
+		-DUSE_SSH=ON \
+		-DCMAKE_C_FLAGS=-fPIC \
+		-DCMAKE_BUILD_TYPE="RelWithDebInfo" \
+		-DCMAKE_INSTALL_PREFIX=$(VENDORDIR)/github.com/libgit2/git2go/$(GIT2GO_VERSION)/static-build/install \
+		-DCMAKE_INSTALL_LIBDIR="lib" \
+		-DDEPRECATE_HARD="${BUILD_DEPRECATE_HARD}" \
+		$(VENDORDIR)/github.com/libgit2/git2go/$(GIT2GO_VERSION)/vendor/libgit2)
+	$(MAKE) -C $(VENDORDIR)/github.com/libgit2/git2go/$(GIT2GO_VERSION)/static-build/build install
+
+$(VENDORDIR)/github.com/libgit2/git2go/$(GIT2GO_VERSION)/vendor/libgit2: $(VENDORDIR)/github.com/libgit2/git2go
+	$(GIT) -C $(VENDORDIR)/github.com/libgit2/git2go/$(GIT2GO_VERSION) submodule update --init --recursive
 
 $(VENDORDIR)/github.com/libgit2/git2go:
-	$(GIT) clone --branch v31.7.9 --recurse-submodules https://github.com/libgit2/git2go.git $@/v31
+	$(GIT) clone --branch $(GIT2GO_VERSION).7.9 --recurse-submodules https://github.com/libgit2/git2go.git $@/$(GIT2GO_VERSION)
```

Comment:
This `GIT2GO_VERSION` feels error prone and very unnecessary to me. For example, there is no guarantee that `v32.7.9` exists here (considering `v31 -> v32`), so the update won't as smooth as bumping this value.

I suggest that we remove this part of the path entirely and vendor the lib in `third_party/libgit2/git2go`.

1. The `replace` in `go.mod` will make it work regardless.
2. The version will be obvious from the `go.mod` file.

Keep it simple, this module is already complex enough.

Reply:
Makes sense, you're right.

I'll simplify the gitmodule in a separate commit and rebase appropriately with your suggested changes.